src/lib.rs

```diff
@@ -476,6 +499,35 @@ pub mod pallet {
 			let _ = frame_system::Pallet::<T>::dec_providers(&supersig_id);
 			Ok(())
 		}
+
+		/// leave a supersig
+		///
+		/// `leave_supersig` will remove caller from selected supersig
+		///
+		/// The dispatch origin for this call must be `Signed` by the user.
+		///
+		/// # <weight>
+		#[transactional]
+		#[pallet::weight(T::WeightInfo::create_supersig())]
+		pub fn leave_supersig(origin: OriginFor<T>, supersig_id: T::AccountId) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			let sindex = Self::get_supersig_index_from_id(&supersig_id)
+				.ok_or(Error::<T>::SupersigNotFound)?;
+
+			if !Self::is_user_in_supersig(sindex, &who) {
+				return Err(Error::<T>::NotMember.into())
+			}
+
+			Supersigs::<T>::mutate(sindex, |wrapped_supersig| {
+				if let Some(supersig) = wrapped_supersig {
+					supersig.members.retain(|memb| memb != &who);
+				}
+			});
+			Self::deposit_event(Event::<T>::SupersigLeaved(supersig_id, who));
+
+			Ok(())
+		}
 	}
 
 	impl<T: Config> Pallet<T> {
```

Comment:
`if index < Self::nonce_supersig()`
Get rid of this check. and start returning an error in the external bodies.
A supersig can be removed from storage. Your check will not know about it, return an index, which will later be unwrapped and the whole chain will crash.

Reply:
Perfect example of what I was saying earlier. You though that you could unwrap and never fail because you did a check in this function. But the code changed, and now this is really dangerous. Just always handle the Result you get